content/en/docs/languages/java/sdk.md

```diff
@@ -0,0 +1,1386 @@
+
+The following code snippet demonstrates `LogRecordProcessor` programmatic
+configuration:
+
+<!-- prettier-ignore-start -->
+<?code-excerpt "src/main/java/otel/LogRecordProcessorConfig.java"?>
+```java
+package otel;
+
+import io.opentelemetry.sdk.logs.LogRecordProcessor;
+import io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+import io.opentelemetry.sdk.logs.export.SimpleLogRecordProcessor;
+import java.time.Duration;
+
+public class LogRecordProcessorConfig {
+  public static LogRecordProcessor batchLogRecordProcessor(LogRecordExporter logRecordExporter) {
+    return BatchLogRecordProcessor.builder(logRecordExporter)
+        .setMaxQueueSize(2048)
+        .setExporterTimeout(Duration.ofSeconds(30))
+        .setScheduleDelay(Duration.ofSeconds(1))
+        .build();
+  }
+
+  public static LogRecordProcessor simpleLogRecordProcessor(LogRecordExporter logRecordExporter) {
+    return SimpleLogRecordProcessor.create(logRecordExporter);
+  }
+}
+```
+<!-- prettier-ignore-end -->
+
+Implement the `LogRecordProcessor` interface to provide your own custom log
+processing logic. For example:
+
+<!-- prettier-ignore-start -->
+<?code-excerpt "src/main/java/otel/CustomLogRecordProcessor.java"?>
+```java
+package otel;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.logs.LogRecordProcessor;
+import io.opentelemetry.sdk.logs.ReadWriteLogRecord;
+
+public class CustomLogRecordProcessor implements LogRecordProcessor {
+
+  @Override
+  public void onEmit(Context context, ReadWriteLogRecord logRecord) {
+    // Callback invoked when log record is emitted.
+    // Enrich the record a custom attribute.
+    logRecord.setAttribute(AttributeKey.stringKey("my.custom.attribute"), "hello world");
+  }
+
+  @Override
+  public CompletableResultCode shutdown() {
+    // Optionally shutdown the processor and cleanup any resources.
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode forceFlush() {
+    // Optionally process any records which have been queued up but not yet processed.
+    return CompletableResultCode.ofSuccess();
+  }
+}
+```
+<!-- prettier-ignore-end -->
+
+#### LogRecordExporter
+
+A
+[LogRecordExporter](https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-logs/latest/io/opentelemetry/sdk/logs/export/LogRecordExporter.html)
+is a [plugin extension interface](#sdk-plugin-extension-interfaces) responsible
+for exporting log records out of process. Rather than directly registering with
+`SdkLoggerProvider`, they are paired with
+[LogRecordProcessors](#logrecordprocessor) (typically
+`BatchLogRecordProcessor`).
+
+Span exporters built-in to the SDK and maintained by community in
```

Comment:
```suggestion
Span exporters that are built-in to the SDK and maintained by the community in
```

Reply:
A long time ago a really good writer drilled a principle into my head to omit any word which isn't explicitly needed. She had this set of common junk words to avoid, which included "that". It turns out sentences using "that" can almost always retain their meaning if you just delete the word, and doing so tends to result in better prose. 

I kept "the" for all the comments like this, but skipped "that are".

Reply:
Makes sense to me. Thanks for sharing that principle, I will also use that approach going forward.